integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, AsyncGenerator, Optional, List
+from httpx import Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from .utils import CustomProperties
+
+PAGE_SIZE = 50
+
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+
+
+class ClickupClient:
+    """Clickup client to interact with Clickup API."""
+
+    def __init__(self, clickup_url: str, clickup_token: str):
+        self.clickup_url = clickup_url
+        self.clickup_token = clickup_token
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.client = http_async_client
+        self.client.timeout = Timeout(30)
+        self.team_key = None
```

Comment:
Why are you tracking team_key?

Reply:
I have fixed the issue by using @cache_iterator_result() to keep the teams in cache